protocols/jvss/jvss.go

```diff
@@ -15,645 +21,249 @@ import (
 	"github.com/dedis/crypto/poly"
 )
 
-// JVSS Protocol Instance structure holding the information for a long-term JVSS
-// signing mechanism
-type JVSSProtocol struct {
-	// The TreeNode denotating ourself in the tree
-	*sda.Node
-	// The EntityList we are using / this is needed to "bypass" the tree
-	// structure for the internals communication, when we set up the shares and
-	// everything. We directly send our share to everyone else directly by using
-	// this entitylist instead of broadcasting into the tree.
-	List *sda.EntityList
-	// the index where we are in this entitylist
-	index int
-
-	// a flat list of all TreeNodes
-	nodeList []*sda.TreeNode
-	// list of public keys represented in the entityList (needed by poly.Deal)
-	publicList []abstract.Point
-	// keys of the Host set as config.KeyPair
-	key config.KeyPair
-
-	info poly.Threshold
-	// The channel where we give the deal we receive for the longterm
-	// generation
-	ltChan chan LongtermChan
-	// The channel through we give the deal we receive for the random generation
-	rdChan chan RandomChan
-	// channel where we give the requests we receive for a signature
-	reqChan chan RequestChan
-	// channel where we give the responses we receive for a signature request
-	respChan chan ResponseChan
-	// requests holds all the requests that we asked
-	requests map[int]*RequestBuffer
-	// lastrequestnumber seen or executed
-	lastRequestNo int
-
-	longterm     *LongtermRequest
-	longtermLock *sync.Mutex
-
-	// callback to know when the longterm has been generated
-	onLongtermDone func(*poly.SharedSecret)
+func init() {
+	sda.ProtocolRegisterName("JVSS", NewJVSS)
 }
 
-type LongtermRequest struct {
-	// The longterm shared private public key pair used in this JVSS.
-	// The idea is that you can keep this protocol instance as a longterm JVSS
-	// using it to distributively sign anything as long as it runs.
-	secret *poly.SharedSecret
-	// The schnorr struct used to sign / verify using the longterm key
-	schnorr *poly.Schnorr
-	// Threshold related to how the shares are generated and reconstructed
-	info poly.Threshold
-	// suite used
-	suite abstract.Suite
-	// key from this node
-	key config.KeyPair
-	// longterm-receiver of the deals
-	receiver *poly.Receiver
-	// doneChan
-	doneChan chan bool
-	// done flag
-	done bool
-	// done lock
-	doneLock *sync.Mutex
-	// how many deals are ok
-	goodDeal int
-}
+// Identifiers for long- and short-term shared secrets.
+const (
+	LTSS = "LTSS"
+	STSS = "STSS"
```

Comment:
Nitpicking: Wouldn't it make sense to make this an enum instead of string constants?
Then you can also refer to the identifiers type where it is needed (for instance in `initSecret `)